app/javascript/controllers/setting_controller.js

```diff
@@ -165,12 +165,10 @@ export default class extends Controller {
     const setDark = function(e) {
       if (e && e.matches) {
         $("body").addClass("night");
+      } else if (isNightMode) {
+        $("body").addClass("night");
       } else {
-        if (isNightMode) {
-          $("body").addClass("night");
-        } else {
-          $("body").removeClass("night");
-        }
+        $("body").removeClass("night");
```

Comment:
if we really have to refactor this, how about following: 

```
if (e && e.matches || isNightMode) {
        $("body").addClass("night");
 } else{
        $("body").removeClass("night");
} 
```

Reply:
Agreed